aspnet/tutorials/web-api-help-pages-using-swagger.rst

```diff
@@ -0,0 +1,536 @@
+
+*Install using NuGet*
+
+.. code-block:: bash
+
+    Install=Package Swashbuckle -Pre
+    
+*Install by modifying project.json*
+
+.. code-block:: javascript
+   :emphasize-lines: 17
+
+    "dependencies": {
+      "Microsoft.NETCore.App": {
+        "version": "1.0.0",
+        "type": "platform"
+      },
+      "Microsoft.ApplicationInsights.AspNetCore": "1.0.0",
+      "Microsoft.AspNetCore.Mvc": "1.0.0",
+      "Microsoft.AspNetCore.Server.IISIntegration": "1.0.0",
+      "Microsoft.AspNetCore.Server.Kestrel": "1.0.0",
+      "Microsoft.Extensions.Configuration.EnvironmentVariables": "1.0.0",
+      "Microsoft.Extensions.Configuration.FileExtensions": "1.0.0",
+      "Microsoft.Extensions.Configuration.Json": "1.0.0",
+      "Microsoft.Extensions.Logging": "1.0.0",
+      "Microsoft.Extensions.Logging.Console": "1.0.0",
+      "Microsoft.Extensions.Logging.Debug": "1.0.0",
+      "Microsoft.Extensions.Options.ConfigurationExtensions": "1.0.0",
+      "Swashbuckle": "6.0.0-beta901"
+
+
+*Install via Manage NuGet Packages*
+
+.. image:: web-api-help-pages-using-swagger/_static/manage-nuget-packages.png
+
+
+*Manage NuGet Packages Tab*
+
+.. image:: web-api-help-pages-using-swagger/_static/add-swagger.png
+
+
+Adding Middleware
+-----------------
+
+After adding the NuGet package to the project, the Startup.cs class needs to modified for the Swagger components to available for use.
+
+First, add SwaggerGen to the services collection in the Configure method.
+
+.. code-block:: c#
+   :emphasize-lines: 12
+
+    public void ConfigureServices(IServiceCollection services)
+    {
+        // Add framework services.
+        services.AddMvc();
+
+        services.AddLogging();
+
+        // Add our repository type
+        services.AddSingleton<ITodoRepository, TodoRepository>();
+
+        // Inject an implementation of ISwaggerProvider with defaulted settings applied
+        services.AddSwaggerGen();
+    }
+
+Next, in the Configure method, enable the middleware for serving generated JSON document and the SwaggerUI. 
+
+.. code-block:: c#
+   :emphasize-lines: 6,9
+
+    public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
+    {
+        app.UseMvcWithDefaultRoute();
+
+        // Enable middleware to serve generated Swagger as a JSON endpoint
+        app.UseSwagger();
+
+        // Enable middleware to serve swagger-ui assets (HTML, JS, CSS etc.)
+        app.UseSwaggerUi();
+        
+    }
+
+In Visual Studio, press ^F5 to launch the app and navigate to ``http://localhost:<random_port>/swagger/v1/swagger.json`` to see the document generated that describes the endpoints.
+
+.. note:: Microsoft Edge, Google Chrome and Firefox display JSON documents natively.  There are extensions for Chrome that will format the document for easier reading.
```

Comment:
They are easy to find, and if they are removed I don't want us to have to be on the hook for updating the tutorial or appear to be supporting one over the other.